docs/guide/performance.md

```diff
@@ -0,0 +1,94 @@
+# Performance
+
+While Vite is fast by default, performance issues can creep in as the project's requirements grow. This guide aims to help you identify and fix common performance issues, such as:
+
+- Slow server starts
+- Slow page loads
+- Slow builds
+
+## Audit Configured Vite Plugins
+
+Vite's internal and official plugins are optimized to do the least amount of work possible. For example, code transformations use regex in dev, but do a complete parse in build to ensure correctness. However, the performance of community plugins is out of Vite's control, which may affect the developer experience.
+
+Here are a few things you can look out for when using additional Vite plugins:
+
+1. The `buildStart`, `config`, and `configResolved` hooks should not run long and extensive operations. These hooks are awaited during dev server startup, which delays when you can access the site in the browser.
+
+2. The `resolveId`, `load`, and `transform` hooks may cause some files to load slower than others. Sometimes this may be unavoidable, but it's worth checking for ways to optimize if possible. For example, checking if the `code` contains a specific keyword, or the `id` matches a specific extension, before doing the full transformation.
+
+   The longer it takes to transform a file, the more significant the request waterfall will be when loading the site in the browser.
+
+   You can inspect the duration it takes to transform a file using `DEBUG="vite:plugin-transform" vite` or [vite-plugin-inspect](https://github.com/antfu/vite-plugin-inspect). Note that as asynchronous operations tend to provide inaccurate timings, you should treat the numbers as a rough estimate, but it should still reveal the more expensive operations.
+
```

Comment:
The get even more information and help the team improve performances of Vite core, you can run `vite dev --profile` and press p & enter when the page is loaded. The profile can be analyzed with a tool like [speedscope](https://www.speedscope.app/).